doc/sphinx/proof-engine/tactics.rst

```diff
@@ -1257,115 +1257,82 @@ Managing the local context
    for any unresolved variables rather than failing.  Typically this happens when
    using a ``%`` intropattern (see :n:`@simple_intropattern`).
 
-.. tacn:: clear @ident
-   :name: clear
+.. tacn:: clear {| {? {? - } {+ @ident } } | @natural }
 
-   This tactic erases the hypothesis named :n:`@ident` in the local context of
-   the current goal. As a consequence, :n:`@ident` is no more displayed and no
-   more usable in the proof development.
+   Erases hypotheses from the context of the current goal, provided that they are
+   not referenced directly or indirectly from the remaining hypotheses or the goal.
+   That means the hypotheses will no longer be shown and can no longer be
+   used in the proof.
 
-   .. exn:: No such hypothesis.
-      :undocumented:
+   If no arguments are given, all hypotheses are erased except for the ones the
+   goal depends on.  If the goal depends on all hypotheses, the context is left
+   unchanged.  Otherwise, the specified hypotheses and all hypotheses that
+   depend on them that the goal doesn't depend on are erased.
```

Comment:
The wording here is a bit hard to read. Perhaps "depend on them which the goal"?